src/main/java/uk/gov/companieshouse/acsp/service/AcspService.java

```diff
@@ -162,7 +175,8 @@ public ResponseEntity<Object> getAcspApplicationStatus(String userId, String req
                 LOGGER.info("No application found for userId: " + userId);
                 return new ResponseEntity<>(HttpStatus.NOT_FOUND);
             }
-            String transactionId = application.get().getLinks().get("self").split("/")[2];
+
+            String transactionId = application.get().getData().getLinks().get("self").split("/")[2];
```

Comment:
I know this is old line... but transactionId can be obtained from transaction object. We dont need to do these many dot operations